src/Tests/Eshopworld.Messaging.Tests/MessengerQueueTest.cs

```diff
@@ -169,6 +171,52 @@ public async Task Test_LockAbandon_MessageFlow()
         }
     }
 
+    [Fact, IsLayer1]
+    public async Task Test_Concurrency_MessageFlow()
+    {
+        await ServiceBusFixture.ServiceBusNamespace.ScorchNamespace();
+
+        using (var ts = new CancellationTokenSource())
+        using (var msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+
+            var sendTasks = new List<Task>();
+            for (var i = 0; i < 2; i++)
+            {
+                var sendTask = msn.Send(new TestMessage());
+                sendTasks.Add(sendTask);
+            }
+
+            var messageCount = 0;
+
+            msn.Receive<TestMessage>(
+                async m =>
+                {
+                    try
+                    {
+                        await msn.Lock(m);
+                        await Task.Delay(TimeSpan.FromSeconds(5), ts.Token);
+                        await msn.Complete(m);
+                        messageCount++;

```

Comment:
You may want to have this increment done as concurrent inc - otherwise there is a chance you will get incorrect value at the end.

Reply:
I have changed this